### ISSUE TITLE
Ignore `wp-settings-*` cookies

### ISSUE THREAD

**Issue by pj-dave** [pressjitsu/pj-page-cache-red#7]
When users logout they'll still see different cache buckets due to the uniqueness of some `wp-settings-*` cookies used only in the admin panel. We can safely ignore these cookies to increase the cache hit-rate for users that may have been logged in at some point.

**Comment by kochtopf** [comment]
Yes found that too and did:

265c274,275
< 			if ( substr( $key, 0, 1 ) === '_' ) {
---
> //			if ( substr( $key, 0, 1 ) === '_' ) {
> if ( substr( $key, 0, 1 ) === '_' || substr( strtolower( $key ), 0, 14 ) === 'comment_author' || substr( strtolower( $key ), 0, 11 ) === 'wp-settings' || substr( strtolower( $key ), 0, 9 ) === 'mindspark' ) {

forget about comment_author here. I've changed the plug to allow a commenter to get cached again after 2 hours of putting a comment. Else commenters will not be cached for 1 year. And the mindspark is just an another cookie spammer I want to get rid of.

**Comment by kochtopf** [comment]
A quick fix for this might be something like this:

> $GLOBALS['wp_filter']['wp_logout'][10]['pj-page-cache'] = array( 'function' => array( **CLASS**, 'wp_logout' ) );
> 
> public static function wp_logout( ) { //Fires after a user is logged-out
>  setcookie( 'wp-settings-' . get_current_user_id(), null, 1 );
>  setcookie( 'wp-settings-time-' . get_current_user_id(), null, 1 );

**Comment by soulseekah** [comment]
Related discussions in Trac:

https://core.trac.wordpress.org/ticket/32567

Let's hope this hits core and we don't have to get our hands dirty like this.

**Comment by kochtopf** [comment]
Yes I understand your pain. I've tried a couple of things and settled for:

265c276,277
< 			if ( substr( $key, 0, 1 ) === '_' ) {
---
> //			if ( substr( $key, 0, 1 ) === '_' ) {
> if ( substr( $key, 0, 1 ) === '_' || substr( strtolower( $key ), 0, 12 ) === 'wp-settings-' || substr( strtolower( $key ), 0, 9 ) === 'mindspark' ) {	//prevent multiple cached pages with the same content but different cookies
431a449
> if ( strpos( strtolower( $cookie_key ), 'wp-settings-' ) !== false ) break;

**Comment by soulseekah** [comment]
Well, that looks overkill and you're messing with the core code of the plugin, meaning you'll need to merge updates. A better approach for now would be to add the following code to your functions.php or something:

```php
add_action( 'clear_auth_cookie', function() {
    setcookie( 'wp-settings-time-' . get_current_user_id(), ' ', time() - YEAR_IN_SECONDS, SITECOOKIEPATH, null ); 
    setcookie( 'wp-settings-' . get_current_user_id(),      ' ', time() - YEAR_IN_SECONDS, SITECOOKIEPATH, null ); 
} );
```

Simpler, no? Doesn't alter code, is future-proof, and can be removed once these cookies are cleared by default (I already submitted a patch here: https://core.trac.wordpress.org/ticket/32567).

**Comment by kochtopf** [comment]
Indeed better. Thank you.

**Comment by soulseekah** [comment]
Fixed upstream in https://core.trac.wordpress.org/changeset/40580 landed in WordPress core 4.8. Closing as fixed.
